### PR TITLE
Switch to Istanbul for coverage

### PR DESCRIPTION
**Pull request by TheTrio**
We currently use [JSCover](https://github.com/tntim96/JSCover) for getting Javascript coverage. Unfortunately, [JSCover doesn't support source maps](https://github.com/tntim96/JSCover/issues/224) which means that to get coverage data anyone can actually read and understand, we split up the `main.js` bundle(generated by webpack) into different files. This is done in `lib/tasks/coverage.rake`

This is problematic for multiple reasons - firstly even a slight change in the output can break it(since it uses regular expressions)- as happened recently when webpack switched to arrow functions instead of the using the `function` keyword. Secondly, the coverage report is based off of the code transpiled by webpack. This not only affects the statistics but more importantly, makes it harder to read the code.

 This PR attempts to switch to [Istanbul](https://istanbul.js.org/) instead, which does support the use of source maps. This means that the hack with the regex and splitting of `main.js` bundle is no longer required. 

Unfortunately, Istanbul [doesn't support storing this coverage data in local storage](https://github.com/istanbuljs/istanbuljs/issues/315). The way I've dealt with this is by writing the coverage report to the disk at the end of each test and merging it with the previous tests results.

However, this means that if the page reloads or refreshes during the test, there would be some loss of data. This is because Istanbul stores the coverage report in `window.__coverage__` instead of a persistent store.

Istanbul also generates considerably larger coverage files - around 6MB, which exceeds the 5MB limit Chrome has for local storage. So even writing to local storage manually is not an option. The only good way I can think of right now is to inject some JS ourselves so that whenever the page changes, it writes the coverage data to Indexed DB, which has a storage limit of 50MB. 

But even that might not be reliable [given how the History API works](https://stackoverflow.com/a/3522154/5055190) - so the most robust(but also the most painful) option might just be to change the way Istanbul works - instead of writing to `window.__coverage__` we write to IndexedDB. But that won't be straightforward either, since IndexedDB is non blocking, not to mention that we'd have to rewrite a lot of existing code.

I have no good answers here. But hopefully, this is a small case which won't affect the data much.

## How to run

I haven't really decided on the workflow for viewing the coverage yet. For now, this is what you need to do.

```
yarn coverage
COVERAGE=true bundle exec rspec spec/ --color --profile --format documentation
```

The generated report is `public/js_coverage/index.html`. You can either open the html file directly or run `rails s` and go to `http://localhost:3000/js_coverage/index.html`

### CODE REVIEW

**Comment by ragesoss** [comment]
This should also delete the binary in `JSCover`.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -68,19 +68,21 @@ jobs:
...
+            yarn coverage
             yarn run test 
       
       - name: JavaScript linting
         run: yarn lint-non-build
-      
-      - name: Ruby rspec test suite
-        run : COVERAGE=true bundle exec rspec spec/ --color --profile --format documentation
+
+      - name: Test & publish code coverage
```
Comment (ragesoss) [inline .github/workflows/ci.yml:77]: I think the previous name is better; the tests being run in this step are much more important than the coverage.

File: app/assets/javascripts/utils/api.js [hunk 2]
```diff
@@ -400,7 +400,7 @@ const API = {
       throw response;
     }
     alert('Students with overdue trainings notified!');
-    return response.json();
+    return response.text();
```
[the remark this answers is not in the document]
Comment (TheTrio) [reply app/assets/javascripts/utils/api.js:403]: I've gone into detail in the [commit message](https://github.com/WikiEducationFoundation/WikiEduDashboard/commit/a9e0be488ffaee2c98e3a11115ae10ca91efd934)

[the remark this answers is not in the document]
Comment (ragesoss) [reply app/assets/javascripts/utils/api.js:403]: Very nice, thanks.

**Pull request merged**